react-cors-spa/src/App.js

```diff
@@ -16,32 +16,53 @@ COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER I
 TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */
 
-import logo from './logo.svg';
-import logoS3 from './logoS3.png';
-import logoCF from './logoCloudFront.png';
-import './App.css';
 import axios from 'axios';
+import { Quotes, Doges } from "./ApiDisplay";
 import { useState, useEffect } from 'react';
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { ReactQueryDevtools } from "@tanstack/react-query-devtools";
 
 // To be replaced by the endpoint of the API deployed through the CloudFormation Template
-const APIEndPoint = 'to be replaced with your api endpoint here'
+const APIEndPoint = 'https://api.quotable.io/random'
+
+// Create a client for QueryClientProvider to use
+const queryClient = new QueryClient()
+
+// Functional component or t/f block
+// If event.path includes 'env' or 'dev'
+// Then return <ReactQueryDevtools />
 
 function App() {
-  return (
-    <div className="App">
-        <header className="App-header">
-          {APIEndPoint.startsWith('http') &&
-            <APIResult />
-          }
-          <img src={logo} className="App-MainLogo" alt="logo" />
-        </header>
-        <p>This react-based application is hosted in an S3 bucket exposed through a CloudFront distribution</p>
-        <div className="logos">
-            <img src={logoS3} className="App-logoR2L" alt="logo S3" />
-            <img src={logoCF} className="App-logoL2R" alt="logo CloudFront" />
+  if (APIEndPoint.includes('dev')) {
+    return (
+      <QueryClientProvider client={queryClient}>
+        <div className="App">
+          <header className="App-header">
+            {APIEndPoint.startsWith('http') &&
```

Comment:
Two things. 

1) We're not looking at the API endpoint because that's hardcoded in L#26. We should be looking at the url in the browser which can be accessed with `location.href`. See: https://developer.mozilla.org/en-US/docs/Web/API/Window/location#value
2) Notice how the if and the else block are 90% the same, the only difference being the dev tools. As a general concept, if you have anything over 20% repeated, you're doing something wrong. In this case, react has some tools for [conditional rendering](https://legacy.reactjs.org/docs/conditional-rendering.html) that we can use to avoid repeating all of the other code.

Reply:
Send me a text on which conditional from the docs you think is best for our scenario and I can give you feedback before you actually code it up.